flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -125,4 +162,118 @@ public void validateColumnListParams(
+                            Collections.singletonList(simplifiedRexNode),
+                            reducedNodes);
+            // check whether period is the unsupported expression
+            if (!(reducedNodes.get(0) instanceof RexLiteral)) {
+                throw new UnsupportedOperationException(
+                        String.format(
+                                "Unsupported time travel expression: %s for the expression can not be reduced to a constant by Flink.",
+                                periodNode));
+            }
+
+            RexLiteral rexLiteral = (RexLiteral) (reducedNodes).get(0);
+            TimestampString timestampString = rexLiteral.getValueAs(TimestampString.class);
+            checkNotNull(
+                    timestampString,
+                    "The time travel expression %s can not reduce to a valid timestamp string. This is a bug. Please file an issue.",
+                    periodNode);
+
+            TableConfig tableConfig = ShortcutUtils.unwrapContext(relOptCluster).getTableConfig();
+            ZoneId zoneId = tableConfig.getLocalTimeZone();
+            long timeTravelTimestamp =
+                    TimestampData.fromEpochMillis(timestampString.getMillisSinceEpoch())
+                            .toLocalDateTime()
+                            .atZone(zoneId)
+                            .toInstant()
+                            .toEpochMilli();
+
+            SchemaVersion schemaVersion = TimestampSchemaVersion.of(timeTravelTimestamp);
+            IdentifierNamespace identifierNamespace = (IdentifierNamespace) ns;
+            IdentifierNamespace snapshotNameSpace =
+                    new IdentifierSnapshotNamespace(
+                            identifierNamespace,
+                            schemaVersion,
+                            ((DelegatingScope) usingScope).getParent());
+            ns = snapshotNameSpace;
+
+            sqlSnapshot.setOperand(
+                    1,
+                    SqlLiteral.createTimestamp(
+                            timestampString,
+                            rexLiteral.getType().getPrecision(),
+                            sqlSnapshot.getPeriod().getParserPosition()));
+        }
+
+        super.registerNamespace(usingScope, alias, ns, forceNullable);
+    }
+
+    /**
+     * For the time travel scenario, we need to build a SchemaVersion based on the Snapshot node for
+     * IdentifierNamespace in order to generate a new namespace.
+     *
+     * <p>In general, the enclosing Node of IdentifierNamespace is usually SqlSnapshot. However, if
+     * we encounter a situation with an "as" operator, we need to identify whether the enclosingNode
+     * is an "as" call and if its first operand is SqlSnapshot.
+     *
+     * @param ns Validator namespace for validator.
+     * @return snapshot node for generating a new namespace for time travel.
+     */
+    private Optional<SqlSnapshot> getTimeTravelNode(SqlValidatorNamespace ns) {
```

Comment:
I think the naming `getTimeTravelNode` is to speicifed, also, I think when it's not for timetravel like lookup join, it will still return a `SqlSnapshot`, Right?
I'd like just rename to `getSnapShotNode`. 
If you think so, aslo, don't forget to update the java docs,